library/objective-c/EnvoyEngineImpl.m

```diff
@@ -29,13 +117,33 @@ - (void)dealloc {
   [[NSNotificationCenter defaultCenter] removeObserver:self];
 }
 
+- (int)registerFilter:(EnvoyHTTPFilter *)filter {
+  // TODO(goaway): Everything here leaks, but it's all be tied to the life of the engine.
+  // This will need to be updated for https://github.com/lyft/envoy-mobile/issues/332
+  ios_http_filter_context *context = safe_malloc(sizeof(ios_http_filter_context));
+  CFBridgingRetain(filter);
+  context->filter = filter;
+  envoy_http_filter *api = safe_malloc(sizeof(envoy_http_filter));
+  api->on_request_headers = ios_http_filter_on_request_headers;
+  api->on_request_data = NULL;
+  api->on_response_headers = ios_http_filter_on_response_headers;
+  api->on_response_data = NULL;
+  api->context = context;
+  register_platform_api(filter.name.UTF8String, api);
```

Comment:
Are we able to generate a filter name here (for the user) instead of requiring the implementer to handle the name for the registration?

Reply:
Yep, that's covered in the dynamic configuration task in the linked issue. @junr03 suggested, and I agreed that we split that into a separate PR.